server/base/src/main/java/org/apache/accumulo/server/master/balancer/HostRegexTableLoadBalancer.java

```diff
@@ -220,7 +220,7 @@ protected String getPoolNameForTable(String tableName) {
    * @param conf
    *          server configuration
    */
-  protected void parseConfiguration(ServerConfiguration conf) {
```

Comment:
> If I'm understanding these changes correctly, it seems the method naming/refactoring isn't the important bit. Was that done for testing purposes or some other reason?
> 
> The most minimal versions of these changes seem to be to simply add two additional `parseConfiguration` calls to the `getAssignments` and `balance` methods. Can we go with the more minimal change without the method renaming/new method?

The reason for the separate methods is because the Observer interface is only for changes to the table configuration and hence updating system properties from those listener methods is not needed.

Reply:
Should we even bother with the observer rebuild, though, if we're calling `parseSystemConfiguration` every time we balance, which calls `parseTableConfiguration` every time? The point of the observer was to avoid reparsing the config unless there was a change. Now, the config is reparsed every time, no matter what, *plus* when the config changes.

Also, note that the observer interface stuff changed in the main branch. And, I believe it may have changed in a way that would work for detecting changes to the system config as well, but I'd have to check.

Reply:
That is a very fair observation.  Thank you.  I will remove the observer mechanism.

Reply:
done

Reply:
Are we okay with the performance hit, if any?

Reply:
Yes